File: backend/src/main/java/com/admin/modules/product/entity/Category.java
package com.admin.modules.product.entity;

import com.admin.common.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "categories")
@Data
@EqualsAndHashCode(callSuper = true)
public class Category extends BaseEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 100)
    private String name;
    
    @Column(nullable = false, unique = true, length = 50)
    private String code;
    
    @Column(length = 500)
    private String description;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Category parent;
    
    @OneToMany(mappedBy = "parent", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Category> children = new ArrayList<>();
    
    @Column(nullable = false)
    private Integer level = 1;
    
    @Column(nullable = false)
    private Integer sortOrder = 0;
    
    @Column(nullable = false)
    private Boolean isActive = true;
    
    @Column(length = 500)
    private String imageUrl;
    
    @OneToMany(mappedBy = "category", fetch = FetchType.LAZY)
    private List<Product> products = new ArrayList<>();
}

File: frontend/src/types/category.ts
// 分类信息接口
export interface Category {
  id: number
  name: string
  description?: string
  parentId: number
  sortOrder: number
  status: number
  createdAt: string
  updatedAt: string
}

// 分类表单接口
export interface CategoryForm {
  name: string
  description?: string
  parentId: number
  sortOrder: number
  status: number
}

// 分类树节点接口
export interface CategoryTreeNode {
  id: number
  name: string
  children?: CategoryTreeNode[]
}

// 分类树节点DTO接口
export interface CategoryTreeNodeDTO {
  id: number
  name: string
  description?: string
  parentId: number
  sortOrder: number
  status: number
  children?: CategoryTreeNodeDTO[]
}